File: Data Structures & Algorithms/counting-bits/submission-0.js
class Solution {
    /**
     * @param {number} n
     * @return {number[]}
     */
    countBits(n) {        
        const res = [0];
        if (n === 0)
            return res;
        res.push(1);
        let cur = 2;

        while (cur <= n) {

            let temp = cur;
            let count = 0;
            while (temp) {
                temp &= temp - 1;
                count++;
            }
            res.push(count);

            cur++;
        }
        return res;
        }
}

File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        if (s.size() != t.size())
            return false;

        insertionSort(s);
        insertionSort(t);

        return s == t;
    }

    void insertionSort(string &s)
    {
        for (size_t i = 1; i < s.size(); ++i)
        {
            int j = i - 1;
            while (j >= 0 && s[j] > s[j + 1])
            {
                auto temp = s[j];
                s[j] = s[j + 1];
                s[j + 1] = temp;
                j--;
            }
        }
    }
};